sdk/inc/azure/iot/az_iot_provisioning_client.h

```diff
@@ -174,6 +188,11 @@ AZ_NODISCARD az_result az_iot_provisioning_client_sas_get_signature(
  * @param[in] mqtt_password_size The size, in bytes of \p mqtt_password.
  * @param[out] out_mqtt_password_length __[nullable]__ Contains the string length, in bytes, of \p
  * mqtt_password. Can be `NULL`.
+ * @pre \p client must not be `NULL`.
+ * @pre \p base64_hmac_sha256_signature must be a valid span of size greater than 0.
```

Comment:
note here that we had switched `base64_hmac_sha256_signature` and `token_expiration_epoch_time` from recommendation of the SDK team so that the two SAS key APIs match (token expiration is second). This was never changed but we might just have to live with it.
https://github.com/Azure/azure-sdk-for-c/blob/66cd00042a12466d998811c9c504615f0d5afd29/sdk/src/azure/iot/az_iot_hub_client_sas.c#L85-L86

Reply:
cc @ericwol-msft

Reply:
Oh interesting, oversight like this can some times happen. I think leaving it as is seems reasonable rather than forcing an API break. There isn't substantial customer value to justify it, but for future APIs let's try to be consistent with the SAS key ones.

Maybe add a document `@remark` to call attention to the parameter order, but we could also leave that alone too, since the compiler should error out if you try to pass an int to the signature parameter.